### ISSUE TITLE
Kodi force close when switching from an add-on to another

### ISSUE THREAD

**Issue by pretoriano80** [xbmc/xbmc#21045]
### Describe the bug
Here is a clear and concise description of what the problem is:

Kodi is force closing when you switch from an add-on to another. 
The involved addons are working correctly when they are used individually and no switch between addons is occurring. 

For this report two well-known addons were use, Netflix addon and Tubed. 

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

The add-on should just work and Kodi shouldn't force close. 

## Actual Behavior

Kodi is force closing 

### To Reproduce

1. Open Netflix addon and navigate to a sub-category (no video needs to be played) 
2.Go back to Video addons section, open Tubed add-on and navigate to a sub-category 
3.Go back to Video addons section, open Netflix again and try to open a sub-category 

After step 3. Kodi should force close, if it does not on the first try, please repeat the procedure. 
For me this pattern would trigger the force close with 99% succes rate. 

Note: The same steps can be reproduced with the Tube add-on too, just start the first step by opening Tube addon, step 2. with Netflix and step 3. with Tubed

Note 2 : Probably this can be reproduced with other addons combo, but for me this had a high rate of reproducibility and i also used both addons frequently. 
### Debuglog

The debuglog can be found here:
https://pastebin.com/yD54bvq0

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

It's a very nasty issue, and WAF is heavily affected. 

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [ ] OSX
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name:
 - Kodi version: Kodi 20 alpha (nightly 25-02-2022)

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by enen92** [comment]
@pretoriano80 debug log link is broken

**Comment by pretoriano80** [comment]
@enen92  sorry for that, pastebin is playing tricks on me today, uploaded directly to GitHub.

**Comment by enen92** [comment]
Are you running latest master? I was pretty sure this crash would be because of 2 concurrent busy dialogs however in such cases there should be a log entry in the log stating this

**Comment by pretoriano80** [comment]
Yes, just checked and this is the installed version - > kodi-20220227-fc22e73e-master-24533-armeabi-v7a.apk.

Also, i can reproduce the issue on LibreELEC, so not limited to Android - > https://github.com/LibreELEC/LibreELEC.tv/issues/6266

**Comment by pretoriano80** [comment]
@enen92  any idea what might cause this issue? It clearly started after the python version bump, but the logs aren't helping.

**Comment by enen92** [comment]
Nop, no idea sorry. Seems to be some python regression

**Comment by pretoriano80** [comment]
Debugging this issue on Android it's harder (at least for me) ,but i did (and others too) managed to get more detailed crash logs on Libreelec,so if anyone is interested here's the link to the bug report from Libreelec --> https://github.com/LibreELEC/LibreELEC.tv/issues/6266

**Comment by pretoriano80** [comment]
On LibreELEC the issue is fixed by reverting this commit --> https://github.com/python/cpython/commit/52937c26adc35350ca0402070160cf6dc838f359

**Comment by CastagnaIT** [comment]
@pretoriano80 on last v20 nightly i am not able to replicate, can you confirm?

**Comment by pretoriano80** [comment]
Yes,everything works now,just tested on latest Kodi nightly (android ).
